library/blocks/in-world-recipes/src/main/java/org/quiltmc/qsl/inworldrecipes/mixin/ItemMixin.java

```diff
@@ -0,0 +1,22 @@
+package org.quiltmc.qsl.inworldrecipes.mixin;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemUsageContext;
+import net.minecraft.util.ActionResult;
+import org.jetbrains.annotations.NotNull;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(Item.class)
+public abstract class ItemMixin {
+	@Inject(method = "useOnBlock", at = @At("HEAD"), cancellable = true)
+	public void qsl$useOnBlockOverride(ItemUsageContext context, CallbackInfoReturnable<ActionResult> cir) {
+		cir.setReturnValue(qsl$useOnBlock(context));
```

Comment:
This affects every item's useOnBlock method that isn't overridden. You want this to not change vanilla behavior unnecessarily.
```suggestion
		ActionResult result = qsl$useOnBlock(context);
		if (result.isAccepted()) {
				cir.setReturnValue(result);
		}
```

Reply:
Also, overriding the handler method was my original intention, but this also works just as well

Reply:
It returns `ActionResult.PASS` by default, which is the same behavior as vanilla (and I doubt this will change, unless Mojang wants to break interacting with _every single block_).

Reply:
I'll allow it to return `null`, which will cause it to defer to the actual method.

Reply:
oops, for some reason I thought it returned `ActionResult.FAIL`. My b

Reply:
I think `ActionResult.PASS` should defer to the actual method tho, and only `ActionResult.SUCCESS` or `ActionResult.FAIL` should cause it to deviate